src/js/background/context-menu.js

```diff
@@ -0,0 +1,371 @@
+      }
+
+      await browser.menus.create(data, relayContextMenus.utils.onCreatedCallback);
+
+      return Promise.resolve(1)
+      
+    },
+    remove: async (id) => {
+      await browser.menus.remove(id);
+    }
+  }, 
+  utils: {
+    getAliases: async () => {
+      if (await getServerStoragePref()) {
+        try {
+          return await getAliasesFromServer();
+        } catch (error) {
+          // API Error — Fallback to local storage
+          const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+          return relayAddresses;
+        }
+      }
+
+      // User is not syncing with the server. Use local storage.
+      const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+      return relayAddresses;
+      
+    },
+    getGeneratedForHistory: async (website) => {
+      const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+      
+      return relayAddresses.some(alias => website === alias.generated_for);
+    },
+    getHostnameFromUrlConstructor: (url) => {
+      const { hostname } = new URL(url);
+      return hostname;
+    },  
+    getMostRecentAliases: (array, domain)=> {
+      // Flipped to match the same order as the dashboard
+      array.reverse();
+
+      // Remove any sites that match the current site (inverse of getSiteSpecificAliases())
+      const filteredAliases = array.filter(alias => alias.generated_for !== domain);
+
+      // Limit to 5
+      return filteredAliases.slice(0, 5);
+    },
+    getSiteSpecificAliases: (array, domain)=> {
+
+      // Flipped to match the same order as the dashboard
+      array.reverse();
+
+      const filteredAliases = array.filter(alias => alias.generated_for === domain);
+
+      // If 5 results for specific domain
+      return filteredAliases.slice(0, 5);
+    },
+    getUserStatus: {
+      canGenerateMoreAliases: async ()=> {
+        const { premium } = await browser.storage.local.get("premium");
+    
+        // Short-circuit if the user is premium.
+        if (premium) return true;
+        
+        const { maxNumAliases } = await browser.storage.local.get("maxNumAliases");
+        const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+
+        return (maxNumAliases - relayAddresses.length) > 0;
+      },
+      getNumberOfAliases: async () => {
+        const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+        return relayAddresses.length;
+      },
+      canUpgradeToPremium: async()=> {
+        const { premium } = await browser.storage.local.get("premium");
+      
+        // Note: If user is already premium, this will return false.
+        return !premium;
+      },
+    },
+    onCreatedCallback: ()=> {
+      // Catch errors when trying to create the same menu twice.
+      // The browser.menus API is limited. You cannot query if a menu item already exists.
+      // The error it throws does not show up to the user.
```

Comment:
What error does it show? We can't `try/catch` it?

Reply:
This is basically a black-hole for errors when trying to make menu items. I pulled it out thinking it wasn't necessary, but was still getting some errors to show up in the console.